modules/core/src/math/matrix/vpMatrix.cpp

```diff
@@ -2921,10 +2920,14 @@ void vpMatrix::insert(const vpMatrix&A, const unsigned int r,
                       const unsigned int c)
 {
   if( (r + A.getRows() ) <= rowNum && (c + A.getCols() ) <= colNum ){
-    // recopy matrix A in the current one, does not call static function to avoid initialisation and recopy of matrix
-    for(unsigned int i=r; i<(r+A.getRows()); i++){
-      for(unsigned int j=c; j<(c+A.getCols()); j++){
-        (*this)[i][j] = A[i-r][j-c];
+    if (A.colNum == colNum && data != NULL && A.data != NULL && A.size() > 0) {
+      memcpy(data+r*colNum, A.data, sizeof(double)*A.size());
+    } else {
+      // recopy matrix A in the current one, does not call static function to avoid initialisation and recopy of matrix
+      for(unsigned int i=r; i<(r+A.getRows()); i++){
+        for(unsigned int j=c; j<(c+A.getCols()); j++){
+          (*this)[i][j] = A[i-r][j-c];
```

Comment:
Yes indeed. It should be possible to improve `vpMatrix::stack()` also. I will continue to push the developments in this branch.